code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/ConsentExtension.java

```diff
@@ -331,45 +286,23 @@ private void shareCurrentConsents(final Event event) {
 		final Map<String, Object> xdmConsents = consentManager.getCurrentConsents().asXDMMap();
 
 		// set the shared state
-		ExtensionErrorCallback<ExtensionError> errorCallback = new ExtensionErrorCallback<ExtensionError>() {
-			@Override
-			public void error(final ExtensionError extensionError) {
-				MobileCore.log(
-					LoggingMode.DEBUG,
-					ConsentConstants.LOG_TAG,
-					String.format(
-						"ConsentExtension - Failed create XDM shared state. Error : %s.",
-						extensionError.getErrorName()
-					)
-				);
-			}
-		};
-
-		getApi().setXDMSharedEventState(xdmConsents, event, errorCallback);
+		getApi().createXDMSharedState(xdmConsents, event);
 
 		// create and dispatch an consent response event
-		final Event responseEvent = new Event.Builder(
+		Event.Builder responseBuilder = new Event.Builder(
 			ConsentConstants.EventNames.CONSENT_PREFERENCES_UPDATED,
-			ConsentConstants.EventType.CONSENT,
-			ConsentConstants.EventSource.RESPONSE_CONTENT
-		)
-			.setEventData(xdmConsents)
-			.build();
-		ExtensionErrorCallback<ExtensionError> dispatchErrorCallback = new ExtensionErrorCallback<ExtensionError>() {
-			@Override
-			public void error(final ExtensionError extensionError) {
-				MobileCore.log(
-					LoggingMode.DEBUG,
-					ConsentConstants.LOG_TAG,
-					String.format(
-						"ConsentExtension - Failed to dispatch %s event: Error : %s.",
-						responseEvent.getName(),
-						extensionError.getErrorName()
-					)
-				);
-			}
-		};
-		MobileCore.dispatchEvent(responseEvent, dispatchErrorCallback);
+			EventType.CONSENT,
+			EventSource.RESPONSE_CONTENT
+		);
+		responseBuilder.setEventData(xdmConsents);
+
+		if (event != null) {
```

Comment:
We can get a null event from onRegister handleInitialization(). Making an event null check here, for covering this case for not to include inResponseToEvent.  
While event is not null, it will include inResponseToEvent.
cc @emdobrin @addb

Reply:
It looks like this method should dispatch an event, but not a response event to the triggering event. The original code called `MobileCore.dispatchEvent` and not `MobileCore.dispatchResponseEvent`. Also, from the method comment, this method dispatches an event to other extensions which may listen for consent updates, and it's not dispatching an event as a response to an API call.
Can you remove this block and not call `inResponseToEvent`?